File: app/src/main/java/com/dumuzeyn/mp3player/LibraryState.java
package com.dumuzeyn.mp3player;

import java.util.ArrayList;
import java.util.HashSet;

/** Mutable library data owned by the activity graph. */
public final class LibraryState {
    final ArrayList<Track> tracks = new ArrayList<>();
    final HashSet<String> favorites = new HashSet<>();
    final ArrayList<Playlist> playlists = new ArrayList<>();
    HomeContent homeContent = HomeContent.empty();
}

File: app/src/main/java/com/dumuzeyn/mp3player/LibraryState.kt
package com.dumuzeyn.mp3player

/** Mutable library data owned by the activity graph. */
class LibraryState {
    @JvmField val tracks = ArrayList<Track>()
    @JvmField val favorites = HashSet<String>()
    @JvmField val playlists = ArrayList<Playlist>()
    @JvmField var homeContent: HomeContent = HomeContent.empty()
}
